### ISSUE TITLE
QT warning on stderr during build: "Note: No relevant classes found. No output generated."

### ISSUE THREAD

**Issue by adeeb10abbas** [teamspatzenhirn/overlay_rviz_plugins#15]
Setup - 
OS: Jammy 22.04
ROS: Humble 

When building with a ROS2 overlay workspace after cloning the git repo, I encounter the following error as a result of `colcon build --symlink-install`

```
--- stderr: rviz_2d_overlay_plugins                               
/home/arrowhead/ros2_ws/src/rviz_2d_overlay_plugins/rviz_2d_overlay_plugins/include/overlay_utils.hpp:0: Note: No relevant classes found. No output generated.
---
Finished <<< rviz_2d_overlay_plugins [9.88s]

```

Any suggestions?

**Comment by ottojo** [comment]
Hi!
I will try to reproduce this later, but i have a few notes already:
* I think your build was successful (as indicated by "Finished", i think colcon would list it as "Failed" or something similar otherwise). Did you try if the plugin was installed and functional despite the note on stderr?
* I have not seen this output before, perhaps it's related to `--symlink-install`? I know that we usually build without that flag.

Also: Could you let me know which compiler you use? I'm pretty sure we only ever tested with clang 14.0.0...

**Comment by ottojo** [comment]
I see this warning/note also occured in the CI build: https://build.ros2.org/job/Hdev__rviz_2d_overlay_plugins__ubuntu_jammy_amd64/2/consoleFull

I guess this must have also occured during our local build, but i know that there are also compiler warnings from clang which have not been resolved yet, so perhaps that just got missed in the noise...

Anyways: Since the binary release seems to work, even with this warning, i suggest you try if it works like that, and i will see if this note can be resolved...

**Comment by adeeb10abbas** [comment]
Hey! @ottojo Thanks a ton, for the speedy fix. In any case, `symlink install` shouldn't affect anything C++ it's for the Python packages in the ws. I will give this a try and see it if everything works well.

**Comment by ottojo** [comment]
I think it was just a header without any QT stuff in it that was given to the MOC. It removes the warning for me, see #16

**Comment by ottojo** [comment]
Please do test if this fixes the issue for you, and feel free to reopen if it doesn't. Also just ping me if you would like me to release a new (binary release) version of the package with this fix...

**Comment by adeeb10abbas** [comment]
Gotcha, thanks! So this is what I run into now while trying to scrape an overlay workspace with the bazel rules we have in [`drake-ros` ](https://github.com/RobotLocomotion/drake-ros/tree/main/bazel_ros2_rules). This happens with the latest commit (a7d7bf082d09de7674c4dc6edb79ae8e853aa250). 

```

--- captured stderr ---
CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_generator_c" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_fastrtps_c"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_introspection_c"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_c" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_generator_cpp" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_fastrtps_cpp"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_introspection_cpp"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_typesupport_cpp" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:19 (add_library):
  Target "empty_using_rviz_2d_overlay_plugins" links to target
  "rviz_2d_overlay_msgs::rviz_2d_overlay_msgs__rosidl_generator_py" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Generate step failed.  Build files cannot be regenerated correctly.
```
Any suggestions are much appreciated! Thanks 
FWIW - CC: @EricCousineau-TRI

**Comment by ottojo** [comment]
I'm not really familiar with the ros/idl/typesupport message generation, and even less so with bazel, so i can't really help with that.

Maybe it's related to this problem?: https://github.com/teamspatzenhirn/rviz_2d_overlay_plugins/issues/14#issuecomment-1380705752

I think that issue was resolved in some other way, but perhaps there is something wrong with the message-package...

**Comment by adeeb10abbas** [comment]
Gotcha, thanks for all the help. I will keep investigating this. Perhaps, opening a new issue regarding this would be more apt?

**Comment by adeeb10abbas** [comment]
Fwiw, Shane's answer on StackOverflow cleared things up for me - https://stackoverflow.com/questions/75151171/unable-to-use-underlay-ros2-installation-with-drake-ros-as-an-external-in-a-baze/75214423#75214423

TLDR: This occurs due to missing find_package() calls in the CMakeLists.txt file of the packages being scraped. Can be solved using `ros2_local_repository` 

I haven't verified the solution just yet but posting it here as I thought it might help someone else!
